### PR TITLE
Support usage of intermediate nodes as labels in training data

### PR DESCRIPTION
**Pull request by adamhadani**
This PR addresses an issue that was raised in #31 - namely, we were not correctly building the per-node training data set (X, y) in a case where some of the training labels given are pointing to intermediate (non-leaf) nodes.

In addition, we add unit-test coverage for these cases to avoid regressions going forward, and make a couple other small updates to remove some warnings that are emitted in unit-tests due to scikit-learn.

### CODE REVIEW

File: sklearn_hierarchical_classification/classifier.py [hunk 1]
```diff
@@ -275,29 +275,44 @@ def _recursive_build_features(self, X, y, node_id, progress):
...
+            # Nb. doing this check explicitly to avoid FutureWarning, see:
+            # https://stackoverflow.com/questions/40659212/futurewarning-elementwise-comparison-failed-returning-scalar-but-in-the-futur
+            return self.graph_.nodes[node_id]["X"]
+
+        indices = np.flatnonzero(y == node_id)
+        X_out = self.graph_.nodes[node_id]["X"] + self._build_features(
+            X=X,
+            y=y,
+            indices=indices,
+        )
```
Comment (adamhadani) [inline sklearn_hierarchical_classification/classifier.py]: ^ this is the crux of the PR.

When we recursively build training data for each node, we "bubble up" labels from child nodes. we need to also add training data labeled with the current node though for cases where it shows up in the training data, which we didnt do before.

File: sklearn_hierarchical_classification/classifier.py [hunk 2]
```diff
@@ -527,7 +544,12 @@ def _base_estimator_for(self, node_id):
         return clone(base_estimator)
 
     def _make_base_estimator(self, node_id):
-        return LogisticRegression()
+        """Create a default base estimator if a more specific one was not chosen by user."""
+        return LogisticRegression(
+            solver="lbfgs",
+            max_iter=1000,
+            multi_class="multinomial",
+        )
```
Comment (adamhadani) [inline sklearn_hierarchical_classification/classifier.py]: unrelated, this gets rid of the ConvergenceWarning I started seeing in unit-tests

File: sklearn_hierarchical_classification/tests/fixtures.py [hunk 3]
```diff
@@ -89,3 +94,29 @@ def make_classifier_and_data(
...
+def make_clothing_graph(root=ROOT):
+    """Create a mock hierarchy of clothing items."""
+    G = DiGraph()
+    G.add_edge(root, "Mens")
+    G.add_edge("Mens", "Shirts")
+    G.add_edge("Mens", "Bottoms")
+    G.add_edge("Mens", "Jackets")
+    G.add_edge("Mens", "Swim")
+
+    return G
```
Comment (adamhadani) [inline sklearn_hierarchical_classification/tests/fixtures.py]: added these fixture factory methods to use the clothing hierarchy that was used to reproduce the bug in #31
